api/envoy/config/listener/v3/quic_config.proto

```diff
@@ -48,4 +65,12 @@ message QuicProtocolOptions {
   // bound by 6000, regardless of this field or how many connections there are.
   google.protobuf.UInt32Value packets_to_read_to_connection_count_ratio = 5
       [(validate.rules).uint32 = {gte: 1}];
+
+  // The implementation of quic::QuicCryptoClientStreamBase used for this listener.
+  // If not specified the :ref:`QUICHE default one<envoy_v3_api_msg_extensions.quic.v3.CryptoServerStreamConfig>` will be used.
+  QuicCryptoStream crypto_stream = 6;
```

Comment:
I'm not super familiar with the way envoy uses protos to configure different behavior, so I'm sorry for being confused. The comment here talks about an implementation of `quic::QuicCryptoClientStreamBase`, which is a C++ class. This makes me think that the crypto_stream field is going to specify a C++ class. However, the default value for this field is `envoy_v3_api_msg_extensions.quic.v3.CryptoServerStreamConfig`, which is a proto message. What is the mapping between messages and C++ classes? I suspect it must have something to do with getAndCheckFactory()?

Reply:
Hmm, the comment is not supper clear. Fixed now.

Envoy extensions are c++ classes like quic::QuicCryptoClientStreamBase. But how we plug them in is via a 1:1 mapped protobuf. I'm trying to configure which c++ classes to use for quic crypto stream and proof source via the configuration of protobuf messages here.

Reply:
And this is handled via static linked-in library.  envoy/registry/registry.h provides interface for that, and getAndCheckFactory() is a wrapper around those interfaces.

Reply:
AH! Thanks, that helps a lot.

Well, other than that the classes in that file read like a randomly generated strings of design patterns :)
* BaseFactoryCategoryRegistry 
* FactoryRegistryProxy
*  InjectFactory